spec/activation_key_spec.rb

```diff
@@ -326,16 +326,15 @@ def hash_diff(h1, h2)
     @activation_key['products'] = []
     @activation_key['contentOverrides'] = [{:contentLabel => 'hello', :name => nil, :value => '123' }]
 
-    lambda {
-      @cp.update_activation_key(@activation_key)
-    }.should raise_exception(RestClient::BadRequest)
-
-
-    @activation_key['contentOverrides'] = [{:contentLabel => nil, :name => 'value', :value => '123' }]
+    output = @cp.update_activation_key(@activation_key)
+    output['products'].should eq(@activation_key['products'])
+    output['name'].should eq(@activation_key['name'])
 
+    @activation_key['contentOverrides'] = [nil]
     lambda {
       @cp.update_activation_key(@activation_key)
-    }.should raise_exception(RestClient::BadRequest)
+    }.should raise_exception(RestClient::InternalServerError)
```

Comment:
This doesn't look right. We should never expect a 500 error. If this is an error we expect we should be throwing a specific relevant exception (such as 400 Bad Request)